<h2>Why is this an issue?</h2>
<p>An array index out-of-bounds panic is a bug class that occurs in Rust when a program tries to access an array element that does not exist. This bug
can cause your program to crash or behave unexpectedly.</p>
<h3>What is the potential impact?</h3>
<p>Issues of this type interrupt the normal execution of a program, causing it to crash or putting it into an inconsistent state. Therefore, this
issue might impact the availability and reliability of your application, or even result in data loss.</p>
<p>If the computation of an index value is tied to user input data, this issue can potentially even be exploited by attackers to disrupt your
application.</p>
<h2>How to fix it</h2>
<p>To fix an array index out of bounds panic in Rust, you should always ensure that you are accessing array elements within the bounds of the
array.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
let x = [1, 2, 3, 4];

x[9]; // Out of bounds indexing
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
let x = [1, 2, 3, 4];

x[0];
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#out_of_bounds_indexing">https://rust-lang.github.io/rust-clippy/master/index.html#out_of_bounds_indexing</a> </li>
</ul>

